### ISSUE TITLE
Providing more than one class results in class duplication

### ISSUE THREAD

**Issue by Dremora** [cibernox/ember-power-select#336]
If I provide more than one class to power-select, e.g.:
```hbs
{{#power-select class="foo bar baz"}}...
```

then the trigger, options and other elements will get the whole class string with `-options` or `-trigger` appended, e.g.:
```hbs
<div class="ember-power-select-trigger foo bar baz-trigger">
```

As a result, all styles of `foo` and `bar` will leak into child elements.

To fix this behaviour, each of the classes from the class string should get the element name appended.

**Comment by cibernox** [comment]
@Dremora Will check tomorrow in the office

**Comment by cah-briangantzler** [comment]
Was looking at this and what should be the actual result. class is the HTML class that you expect to be on the outer div. Using it to do the -trigger / -option class is probably incorrect as this points out. 

There is already a triggerClass property, I would think that should be used and {{class}}-trigger be removed.

Should an optionsClass property be created and used in place of {{class}}-option

I don't think applying -trigger / -option to each individual class is the correct thing to do nor should class be copied down to trigger and option appended.

**Comment by cibernox** [comment]
Tomorrow I'll think about the problem a bit more, but my intuition is gives me 2 solutions:

1) Remove `class=` entirely and leave just `triggerClass` and `dropdownClass`. More verbose, less magical. 

2) Probably when the users do `class=something` they expect that class to apply to the entire component. Given the weird nature of this component where part of it is rendered in the root of the body, the closest thing we can do is to apply that class/classes to both the trigger and the dropdown, so they can target anything inside it with styles like `.big-dropdown .ember-power-select-option`. Doing `class="foo bar"`will result in both the trigger and the dropdown having `"foo bar"` among their classes.

**Comment by Dremora** [comment]
I think that 2) still leaves a lot of confusion. `class` is a standard HTML attribute, and it means “add this class to this element”. No other behaviour expected. I do like the idea of having two explicit properties for trigger class and options class, and no concatenation magic; it also allows us to generate class names using `ember-css-modules`. Perhaps we should still keep the `class` attribute - we do use it in a few places in our app - but remove the special behaviour.

**Comment by cibernox** [comment]
Closed by #348
